enterprise/internal/codemonitors/resolvers/resolvers.go

```diff
@@ -2,82 +2,256 @@ package resolvers
+	if n == 0 {
+		return nil
+	}
+	return &n
+}
+
+var monitorColumns = []*sqlf.Query{
+	sqlf.Sprintf("cm_monitors.id"),
+	sqlf.Sprintf("cm_monitors.created_by"),
+	sqlf.Sprintf("cm_monitors.created_at"),
+	sqlf.Sprintf("cm_monitors.changed_by"),
+	sqlf.Sprintf("cm_monitors.changed_at"),
+	sqlf.Sprintf("cm_monitors.description"),
+	sqlf.Sprintf("cm_monitors.enabled"),
+	sqlf.Sprintf("cm_monitors.namespace_user_id"),
+	sqlf.Sprintf("cm_monitors.namespace_org_id"),
+}
+
+func monitorsQuery(userID int32, args *graphqlbackend.ListMonitorsArgs) (*sqlf.Query, error) {
+	const SelectMonitorsByOwner = `
+SELECT id, created_by, created_at, changed_by, changed_at, description, enabled, namespace_user_id, namespace_org_id 
+FROM cm_monitors
+WHERE namespace_user_id = %s
+AND id > %s
+ORDER BY id ASC
+LIMIT %S
+`
+	var after int64
+	if args.After == nil {
+		after = 0
+	} else {
+		err := relay.UnmarshalSpec(graphql.ID(*args.After), &after)
+		if err != nil {
+			return nil, err
+		}
+	}
+	query := sqlf.Sprintf(
+		SelectMonitorsByOwner,
+		userID,
+		after,
+		args.First,
+	)
+	return query, nil
+}
+
+func (r *Resolver) createCodeMonitorQuery(ctx context.Context, args *graphqlbackend.CreateCodeMonitorArgs) (*sqlf.Query, error) {
+	const InsertCodeMonitorQuery = `
+INSERT INTO cm_monitors 
+(created_at, created_by, changed_at, changed_by, description, namespace_user_id, namespace_org_id) 
+VALUES (%s,%s,%s,%s,%s,%s,%s)
+RETURNING %s;
+`
+	var userID int32
+	var orgID int32
+	err := graphqlbackend.UnmarshalNamespaceID(args.Namespace, &userID, &orgID)
+	if err != nil {
+		return nil, err
+	}
+	now := r.clock()
+	a := actor.FromContext(ctx)
+	return sqlf.Sprintf(
+		InsertCodeMonitorQuery,
+		now,
+		a.UID,
+		now,
+		a.UID,
+		args.Description,
+		nilOrInt32(userID),
+		nilOrInt32(orgID),
+		sqlf.Join(monitorColumns, ", "),
+	), nil
 }
 
 //
 // MonitorConnection
 //
 type monitorConnection struct {
-	userID graphql.ID
+	*Resolver
+	monitors []graphqlbackend.MonitorResolver
 }
 
 func (m *monitorConnection) Nodes(ctx context.Context) ([]graphqlbackend.MonitorResolver, error) {
-	return []graphqlbackend.MonitorResolver{&monitor{userID: m.userID}}, nil
+	return m.monitors, nil
 }
 
 func (m *monitorConnection) TotalCount(ctx context.Context) (int32, error) {
-	return 1, nil
+	return int32(len(m.monitors)), nil
 }
 
 func (m *monitorConnection) PageInfo(ctx context.Context) (*graphqlutil.PageInfo, error) {
-	return graphqlutil.HasNextPage(false), nil
+	if len(m.monitors) == 0 {
+		return graphqlutil.HasNextPage(false), nil
+	}
+	return graphqlutil.NextPageCursor(string(m.monitors[len(m.monitors)-1].ID())), nil
 }
 
 //
 // Monitor
 //
 type monitor struct {
-	userID graphql.ID
-}
+	id              int64
+	createdBy       int32
+	createdAt       time.Time
+	changedBy       int32
+	changedAt       time.Time
+	description     string
+	enabled         bool
+	namespaceUserID *int32
+	namespaceOrgID  *int32
+}
+
+const (
+	monitorKind = "codemonitor"
```

Comment:
I think the others we have are all capitalized, but not sure it matters.
```suggestion
	monitorKind = "CodeMonitor"
```